plugins/scaffolder-backend-module-github/src/actions/inputProperties.ts

```diff
@@ -311,6 +311,615 @@ const customProperties = {
+            actor_type: {
+              title: 'Actor Type',
+              description: 'The type of actor that can bypass a ruleset.',
+              enum: [
+                'OrganizationAdmin',
+                'Integration',
+                'RepositoryRole',
+                'Team',
+                'DeployKey',
+              ],
+              type: 'string',
+            },
+            bypass_mode: {
+              title: 'Bypass Mode',
+              description:
+                'When the specified actor can bypass the ruleset. `pull_request` means that an actor can only bypass rules on pull requests. `pull_request` is not applicable for the `DeployKey` actor type. Also, `pull_request` is only applicable to branch rulesets. Default: `always`',
+              enum: ['always', 'pull_request'],
+              type: 'string',
+            },
+          },
+          required: ['actor_type', 'bypass_mode'],
+          type: 'object',
+        },
+        type: 'array',
+      },
+      conditions: {
+        title: 'Ruleset Conditions',
+        description: 'Parameters for a repository ruleset ref name condition',
+        additionalProperties: false,
+        properties: {
+          ref_name: {
+            title: 'Ref Name conditions',
+            description:
+              'Parameters for a repository ruleset ref name condition',
+            additionalProperties: false,
+            properties: {
+              exclude: {
+                title: 'Exclude',
+                description:
+                  'Array of ref names or patterns to include. One of these patterns must match for the condition to pass. Also accepts `~DEFAULT_BRANCH` to include the default branch or `~ALL` to include all branches.',
+                items: {
+                  type: 'string',
+                },
+                type: 'array',
+              },
+              include: {
+                title: 'Include',
+                description:
+                  'Array of ref names or patterns to exclude. The condition will not pass if any of these patterns match.',
+                items: {
+                  type: 'string',
+                },
+                type: 'array',
+              },
+            },
+            type: 'object',
+          },
+        },
+        type: 'object',
+      },
+      enforcement: {
+        title: 'Ruleset Enforcement',
+        description:
+          'The enforcement level of the ruleset. evaluate allows admins to test rules before enforcing them. Admins can view insights on the Rule Insights page (`evaluate` is only available with GitHub Enterprise).',
+        enum: ['active', 'disabled', 'evaluate'],
+        type: 'string',
+      },
+      name: {
+        title: 'Ruleset Name',
+        description: `The name of the ruleset.`,
+        type: 'string',
+      },
+      rules: {
+        title: 'Ruleset Rules',
+        description: 'An array of rules within the ruleset.',
+        items: {
+          anyOf: [
```

Comment:
this array could be outdated and hard to track changes.

this list was extracted form Octokit types, but looking in the docs there are some new possible rules. not sure if it is possible to automate this schema from octokit instead